Jenkinsfile

```diff
@@ -9,32 +9,42 @@ stage('Initialize') {
 
 provision(devEnvironmentName())
 if (isMaster()) {
-    input 'Deploy to Production'
     provision("prod")
 }
 
 def provision(environment) {
-    stage('Provision Infrastructure Stack') {
+    def label = ((environment.startsWith("dev")) ? "dev" : environment).
+        toUpperCase();
+
+    if (label != "DEV") {
+        stage("master") {
+            timeout(time:5, unit:'DAYS') {
```

Comment:
Yep, 5 days is arbitrary, although I simply copied it from [Top 10 Best Practices for Jenkins Pipeline Plugin](https://www.cloudbees.com/blog/top-10-best-practices-jenkins-pipeline-plugin). Likely a smaller number makes more sense, but it can always be adjusted later.